examples/distributed/pyg/distributed_cpu.py

```diff
@@ -0,0 +1,497 @@
+    graph.meta = feature.meta = meta
+    feature.labels = torch.load(node_label_file)
+    partition_data = (feature, graph)
+
+    # Initialize distributed context
+    current_ctx = DistContext(
+        world_size=num_nodes,
+        rank=node_rank,
+        global_world_size=num_nodes,
+        global_rank=node_rank,
+        group_name='distributed-sage-supervised-Node',
+    )
+    current_device = torch.device('cpu')
+
+    print('--- Initialize DDP training group ...')
+    # Initialize DDP training process group.
+    torch.distributed.init_process_group(
+        backend='gloo',
+        rank=current_ctx.rank,
+        world_size=current_ctx.world_size,
+        init_method='tcp://{}:{}'.format(master_addr, ddp_port),
+    )
+    num_neighbors = [int(i) for i in num_neighbors.split(',')]
+    persistent_workers = (True if num_workers > 0 else False
+                          )  # Keep workers RPC alive outside the iterator loop
+    print('--- Initialize distributed loaders ...')
+    # Create distributed neighbor loader for training
+    train_loader = pyg_dist.DistNeighborLoader(
+        data=partition_data,
+        input_nodes=train_idx,
+        current_ctx=current_ctx,
+        device=current_device,
+        num_neighbors=num_neighbors,
+        shuffle=True,
+        drop_last=True,
+        persistent_workers=persistent_workers,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        master_addr=master_addr,
+        master_port=train_loader_port,
+        concurrency=concurrency,
+        async_sampling=async_sampling,
+    )
+    # Create distributed neighbor loader for testing.
+    test_loader = pyg_dist.DistNeighborLoader(
+        data=partition_data,
+        input_nodes=test_idx,
+        current_ctx=current_ctx,
+        device=current_device,
+        num_neighbors=num_neighbors,
+        shuffle=False,
+        drop_last=True,
+        persistent_workers=persistent_workers,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        master_addr=master_addr,
+        master_port=test_loader_port,
+        concurrency=concurrency,
+        async_sampling=async_sampling,
+    )
+
+    print('--- Initialize model ...')
+    # Define model and optimizer.
+    model = GraphSAGE(
+        in_channels=128 if is_hetero else 100,  # num_features
+        hidden_channels=256,
+        num_layers=len(num_neighbors),
+        out_channels=349 if is_hetero else 47,  # num_classes in dataset
+    ).to(current_device)
+
+    @torch.no_grad()
+    def init_params():
+        # Init parameters via forwarding a single batch to the model:
+        print('--- Initialize parameters of the hetero model ...')
+        with train_loader as iterator:
+            batch = next(iter(iterator))
+            batch = batch.to(current_device, 'edge_index')
+            model(batch.x_dict, batch.edge_index_dict)
+            del batch
+            torch.distributed.barrier()
```

Comment:
This doesn't look required.

Reply:
It would only be required in case of lazy initialization (`-1` as feature), which I don't find here.

Reply:
You're right. Thanks for pointing that out. TIL